### PR TITLE
server/node: rename Node to MultiRaftServer

### PR DESCRIPTION
**Pull request by zhangjinpeng87**
### What is changed and how it works?

Issue Number: Close https://github.com/tikv/tikv/issues/16782

What's Changed:

Rename `Node` to a better name `MultiRaftServer`.

### Related changes

- None

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- None

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by LykxSassinator** [comment]
IMO, `RaftStore` sounds better than `RaftServer `for TiKV, considering TiKV's multiple raft groups.

Referring to a single `RaftServer` may give the impression that it serves only one raft group.

**Review by tonyxuqqi — APPROVED** [review]
or maybe Raftstore as the name?

**Comment by zhangjinpeng87** [comment]
> IMO, `RaftStore` sounds better than `RaftServer `for TiKV, considering TiKV's multiple raft groups.
> 
> Referring to a single `RaftServer` may give the impression that it serves only one raft group.

`RaftStore` is a name that vague, instead I prefer `MultiRaftServer` to express the `MultiRaft` semantic.

**Review by SpadeA-Tang — APPROVED** [review]

File: src/server/raft_server.rs [hunk 1]
```diff
@@ -92,7 +92,7 @@ pub(crate) fn init_store(store: Option<metapb::Store>, cfg: &ServerConfig) -> me
 
 /// A wrapper for the raftstore which runs Multi-Raft.
 // TODO: we will rename another better name like RaftStore later.
```
Comment (LykxSassinator) [inline src/server/raft_server.rs:94]:
```suggestion
```

**Review by LykxSassinator — COMMENTED** [review]
Rest LGTM

**Pull request merged**